Add configuration samples to Inventory project

bronsh opened the pull request:
Add an example of using configuration with Inventory project

Commit:
Add configuration samples to Inventory project

Commit by David Bronshtein <david.br@gigya-inc.com>
On branch: refs/heads/with_config

Allon-Guralnek commented on InventoryService/InventoryService.Grains/InventoryConfig.cs:
```diff
@@ -0,0 +1,23 @@
+
+namespace InventoryService.Grains
+{
+    [ConfigurationRoot("Inventory", RootStrategy.ReplaceClassNameWithPath)]
```
For simplicity, we should not include the `ConfigurationRoot` attribute in this sample.

Allon-Guralnek commented on InventoryService/InventoryService.Grains/InventoryConfig.cs:
```diff
@@ -0,0 +1,23 @@
+        /// <summary>
+        /// Send warning if stock is lower than this quantity
+        /// </summary>
+        public uint LowStockWarning { get; set; }
```
`LowStockWarningQuantity`

Allon-Guralnek commented on InventoryService/InventoryService/Program.cs:
```diff
@@ -1,16 +1,31 @@
-            Environment.SetEnvironmentVariable("GIGYA_CONFIG_ROOT", Environment.CurrentDirectory);
-            Environment.SetEnvironmentVariable("GIGYA_CONFIG_PATHS_FILE", "");
-            Environment.SetEnvironmentVariable("GIGYA_ENVVARS_FILE", Environment.CurrentDirectory);
+            // Environment variables can be set either by EnvironmentVariables file, or manually, by code.
```
`environmentVariables.json` + via OS

Allon-Guralnek commented on InventoryService/InventoryService/config/global/inventory.config:
```diff
@@ -0,0 +1,17 @@
+  -->
+
+  <Inventory>
+    <MaxItemsInStock>10</MaxItemsInStock>
```
Let's use XML attributes for primitive values, I think it looks better.

Allon-Guralnek commented on InventoryService/InventoryService/config/prod/inventory.config:
```diff
@@ -0,0 +1,13 @@
+    The LowStockWarning entry is not changed, and will have the default value as configured for global level
+  -->
+
+  <Inventory.MaxItemsInStock>6000</Inventory.MaxItemsInStock>  
```
Attribute here too.

Allon-Guralnek commented on InventoryService/InventoryService/Program.cs:
```diff
@@ -1,16 +1,31 @@
+            // The following two rows set variables manually, but actually it is not needed, 
+            // because variables which were not set before, will be set anyway by EnvironmentVarialbes file
+            Environment.SetEnvironmentVariable("GIGYA_CONFIG_ROOT", Path.Combine(Environment.CurrentDirectory, "config"));
+            Environment.SetEnvironmentVariable("GIGYA_CONFIG_PATHS_FILE", "./config/loadPaths.json");
```
file should be named `configPaths.json` to match env var name

Allon-Guralnek commented on InventoryService/InventoryService.Grains/InventoryConfig.cs:
```diff
@@ -0,0 +1,23 @@
+        /// <summary>
+        /// Max number of items which can be stored on Stock
+        /// </summary>
+        public uint MaxItemsInStock { get; set; }
```
`MaxQuantityInStock`

Commit:
change config field names, don't use EnvironmentVariables.json

Commit by David Bronshtein <david.br@gigya-inc.com>
On branch: refs/heads/with_config

Commit:
Added log info about current quantity and max quantity on Stock

Commit by David Bronshtein <david.br@gigya-inc.com>
On branch: refs/heads/with_config
